src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,167 @@
+import {
+  ChangeEvent, KeyboardEvent, useEffect, useRef, useState,
+} from 'react';
+/* eslint-disable quote-props */
+import classNames from 'classnames';
+import { Todo } from '../../types';
+import { useTodos } from '../../context';
+
+type Props = {
+  todo: Todo,
+};
+
+export const TodoItem = ({ todo }: Props) => {
+  const { completed, title, id } = todo;
+
+  const [newtitle, setNewTitle] = useState<string | undefined>(title);
```

Comment:
why it can be undefiend?

Reply:
Because at some stage it was necessary to change the type to 'undefined' and if it is a prop it could be 'undefined'. But now, I see, that, I don't need string | undefined anymore, so done